akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -79,6 +80,60 @@ object Backoff {
+   *
+   * When the `Restart` directive is specified, the supervisor will delay the restart
+   * using an exponential back off strategy (bounded by minBackoff and maxBackoff).
+   *
+   * This supervisor is intended to be transparent to both the child actor and external actors.
+   * Where external actors can send messages to the supervisor as if it was the child and the
+   * messages will be forwarded. And when the child is `Terminated`, the supervisor is also
+   * `Terminated`.
+   * Transparent to the child means that the child does not have to be aware that it is being
+   * supervised specifically by this actor. Just like it does
+   * not need to know when it is being supervised by the usual implicit supervisors.
+   * The only caveat is that the `ActorRef` of the child is not stable, so any user storing the
+   * `sender()` `ActorRef` from the child response may eventually not be able to communicate with
+   * the stored `ActorRef`. In general all messages to the child should be directed through this actor.
+   *
+   * An example of where this supervisor might be used is when you may have an actor that is
+   * responsible for continuously polling on a server for some resource that sometimes may be down.
+   * Instead of hammering the server continuously when the resource is unavailable, the actor will
+   * be restarted with an exponentially increasing back off until the resource is available again.
+   *
+   * '''***
+   * This supervisor should not be used with `Akka Persistence` child actors.
+   * `Akka Persistence` actors shutdown unconditionally on `persistFailure()`s rather
+   * than throw an exception on a failure like normal actors.
+   * [[#onStop]] should be used instead for cases where the child actor
+   * terminates itself as a failure signal instead of the normal behavior of throwing an exception.
+   * ***'''
+   * You can define another
+   * supervision strategy by using `akka.pattern.BackoffOptions.withSupervisorStrategy` on [[akka.pattern.BackoffOptions]].
+   *
+   * @param childProps the [[akka.actor.Props]] of the child actor that
+   *   will be started and supervised
+   * @param childName name of the child actor
+   * @param minBackoff minimum (initial) duration until the child actor will
+   *   started again, if it is terminated
+   * @param maxBackoff the exponential back-off is capped to this duration
+   * @param randomFactor after calculation of the exponential back-off an additional
+   *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
+   *   In order to skip this additional delay pass in `0`.
+   */
+  def onFailure(
+    childProps:   Props,
+    childName:    String,
+    minBackoff:   java.time.Duration,
+    maxBackoff:   java.time.Duration,
+    randomFactor: Double): BackoffOptions =
+    BackoffOptionsImpl(RestartImpliesFailure, childProps, childName, minBackoff.asScala, maxBackoff.asScala, randomFactor)
```

Comment:
call the other `onFailure` method instead

Reply:
ok, it is also an option and i will do it